src/main/java/com/broadleafcommerce/autoconfigure/HSQLDBServer.java

```diff
@@ -64,20 +68,62 @@ public HSQLDBServer(final HSQLDBProperties autoProps, Environment environment) {
 
     @Override
     public boolean isRunning() {
-        try (Socket ignored = new Socket(InetAddress.getByName(null), props.getIntegerProperty("server.port", 0))) {
-            return true;
-        } catch (IOException ignored) {
-            return false;
+        boolean isRunning = false;
+        Connection c = null;
+
+        final int port = props.getIntegerProperty("server.port", 0);
+        try {
+            Class.forName("org.hsqldb.jdbc.JDBCDriver");
+            final String url = "jdbc:hsqldb:hsql://" + InetAddress.getByName(null).getHostName() + ":"
+                               + port + "/"
+                               + props.getProperty("server.dbname.0", "");
+            final String username = "SA";
+            final String password = "";
+            
+            c = DriverManager.getConnection(url, username, password);
+            
+            isRunning = true;
+        } catch (ClassNotFoundException e) {
+            LOG.error("ERROR: Cannot test if an HSQL server is running: failed to load HSQLDB JDBC driver.", e);
+        } catch (SQLException e) {
+            Socket s = null;
+            try {
+                // see if the port is being used already (by something other than HSQL)
+                s = new Socket(InetAddress.getByName(null), port);
+                LOG.error("Port," + port + ", is already in use but not by HSQL");
```

Comment:
Right. Just trying to solve the _next_ problem they will have. The next thing anybody would do at this point would Google "How do I find out what process is running on a port?" and if we put that front and center it gives them a better experience overall. Basically just think what _you_ would want to see if you got this error. What would be most helpful to you? (and think about windows/linux/macos).